### PR TITLE
Code Quality: Introduced SideBarContext

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

Introduced SideBarContext, which helps you move all context menu ICommands to RichCommands.

## Task Checklist

- [x] Introduced SideBarContext
- [x] De-cluttered SideBarViewModel

## Known Issues

_N/A_

## PR Checklist

- [ ] **Closes**: _N/A_
- [x] **Approval**: Have discussed with and got approval from the team[^1]
- [x] **Tests**: Tested accessibility on the local end
- [x] **Deployment**: Deployed the app on the local end
- [ ] **Localization**: Modified en-US string resources[^2]
- [ ] **Co-authors**: _N/A_

## Steps To Validate Changes

No affect to the UI codebase.

## Screenshots

_N/A_

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do that, instead.

### CODE REVIEW

File: src/Files.App/Data/Contexts/SideBar/ISideBarContext.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	{
+		/// <summary>
+		/// The last right clicked item
+		/// </summary>
+		ILocatableSideBarItem? RightClickedItem { get; }
+
+		/// <summary>
+		/// Tells whether any item has been right clicked
+		/// </summary>
+		bool IsAnyItemRightClicked { get; }
```
Comment (yaira2) [inline src/Files.App/Data/Contexts/SideBar/ISideBarContext.cs:16]:
```suggestion
		bool IsItemRightClicked { get; }
```

File: src/Files.App/Views/Properties/GeneralPage.xaml.cs [hunk 2]
```diff
@@ -77,7 +77,7 @@ bool GetNewName(out string newName)
 				return false;
 			}
 
-			bool SaveDrive(DriveItem drive)
+			bool SaveDrive(LocatableDriveItem drive)
```
Comment (yaira2) [inline src/Files.App/Views/Properties/GeneralPage.xaml.cs:80]: What does `Locatable` represent?

Comment (0x5bfa) [reply src/Files.App/Views/Properties/GeneralPage.xaml.cs:80]: The storage abstraction we are going to introduce has ILocatable class and this class implements that class. Locatable means that the class has path, name, and other info, which are all necessary when you are searching the folder on the filesystem.

Comment (yaira2) [reply src/Files.App/Views/Properties/GeneralPage.xaml.cs:80]: @d2dyno1 are you on board with this?

Comment (0x5bfa) [reply src/Files.App/Views/Properties/GeneralPage.xaml.cs:80]: Renamed followings to have Locatable.

- ILocatableSideBarItem
- LocatableWSLItem
- LocatableDriveItem
- LocatableStorageItem

File: src/Files.App/Data/Contexts/SideBar/ISideBarContext.cs (hunk 1 above)
Comment (0x5bfa) [reply src/Files.App/Data/Contexts/SideBar/ISideBarContext.cs:16]: Done

**Pull request closed without merging**